### PR TITLE
add basic debugging guide to production docs

### PR DESCRIPTION
**Pull request by sw-yx**
following on from Ryland's video guide https://youtu.be/PqcVKIxI0nU

deploy preview: https://deploy-preview-295--mystifying-fermi-1bc096.netlify.app/docs/server-production-deployment#debugging-workflows

### CODE REVIEW

File: docs/server-production-deploy.md [hunk 1]
```diff
@@ -49,17 +49,64 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+
+#### Execution Histories on Temporal Web
+
+Reading execution histories is one of the more reliable ways of debugging:
+
+![image](https://user-images.githubusercontent.com/6764957/110546362-54d92c00-8169-11eb-81a6-74817e0d1378.png)
+
+Here, you can see the exact sequence of events that have happened so far, together with the relevant state for each event, and reason about what went wrong or what is preventing the next correct event. *There are about 40 system events in total - We intend to publish a full guide to each.*
+
+
```
Comment (sw-yx) [inline docs/server-production-deploy.md:83]:
```suggestion

```

File: docs/server-production-deploy.md [hunk 2]
```diff
@@ -49,17 +49,62 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+Here, you can see the exact sequence of events that have happened so far, together with the relevant state for each event, and reason about what went wrong or what is preventing the next correct event. *There are about 40 system events in total - We intend to publish a full guide to each.*
+#### Viewing Stack Traces on Temporal Web
+
+Temporal also stores the stack trace of where a given activity is currently blocked:
+
+![image](https://user-images.githubusercontent.com/6764957/110547621-20ff0600-816b-11eb-84f3-c6a97c5cad31.png)
+
+This is often a good way to get a deep understanding of whether your workflow is executing as expected.
+
+
```
Comment (sw-yx) [inline docs/server-production-deploy.md:90]:
```suggestion

```

File: docs/server-production-deploy.md [hunk 3]
```diff
@@ -49,17 +49,60 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+Here, you can see the exact sequence of events that have happened so far, together with the relevant state for each event, and reason about what went wrong or what is preventing the next correct event. *There are about 40 system events in total - We intend to publish a full guide to each.*
+#### Viewing Stack Traces on Temporal Web
+
+Temporal also stores the stack trace of where a given activity is currently blocked:
+
+![image](https://user-images.githubusercontent.com/6764957/110547621-20ff0600-816b-11eb-84f3-c6a97c5cad31.png)
+
+This is often a good way to get a deep understanding of whether your workflow is executing as expected.
+#### Recovering In-flight Workflows While Running
+
```
Comment (sw-yx) [inline docs/server-production-deploy.md:90]:
```suggestion

#### Recovering In-flight Workflows While Running

```

File: docs/server-production-deploy.md [hunk 4]
```diff
@@ -49,17 +49,61 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+If you have the time, we recommend [watching our 19 minute video guide on YouTube](https://youtu.be/PqcVKIxI0nU) which demonstrates the debugging explained below.
+
+#### Basic Debugging on Temporal Web
+
+The primary mechanism we recommend for debugging is Temporal Web:
+
+![6XkjmR](https://user-images.githubusercontent.com/6764957/110544958-71746480-8167-11eb-8152-8d3a3eb73d4e.gif)
+
+- Workflows are identified by their **Workflow ID**, which you provide when creating the workflow. They also have a **Name** which is directly taken from your code.
+- Workflow ID's are are distinct from **Run ID's**, which identify one of potentially many unique runs of a Workflow.
```
Comment (sw-yx) [inline docs/server-production-deploy.md:65]:
```suggestion
- Workflow ID's are are distinct from **Run ID's**, which uniquely identify one of potentially many runs of the same Workflow.
```

File: docs/server-production-deploy.md [hunk 5]
```diff
@@ -49,17 +49,61 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
 
+### Debugging Workflows
+
+> ⚠️ This is a basic guide to troubleshooting/debugging Temporal applications. It is work-in-progress and we encourage [reading about our Architecture](https://docs.temporal.io/docs/server-architecture) for more detail. The better you understand how Temporal works, the better you will be at debugging your workflows.
+
+If you have the time, we recommend [watching our 19 minute video guide on YouTube](https://youtu.be/PqcVKIxI0nU) which demonstrates the debugging explained below.
+
+#### Basic Debugging on Temporal Web
+
+The primary mechanism we recommend for debugging is Temporal Web:
```
Comment (sw-yx) [inline docs/server-production-deploy.md:60]:
```suggestion
The primary mechanism we recommend for debugging is [Temporal Web](https://github.com/temporalio/web), which is run in a separate process:
```

File: docs/server-production-deploy.md [hunk 6]
```diff
@@ -49,17 +49,61 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+
+![6XkjmR](https://user-images.githubusercontent.com/6764957/110544958-71746480-8167-11eb-8152-8d3a3eb73d4e.gif)
+
+- Workflows are identified by their **Workflow ID**, which you provide when creating the workflow. They also have a **Name** which is directly taken from your code.
```
Comment (sw-yx) [inline docs/server-production-deploy.md:64]:
```suggestion
- Workflows are identified by their [**Workflow ID**](https://docs.temporal.io/docs/glossary/#workflow-id), which you provide when creating the workflow. They also have a **Name** which is directly taken from your code.
```

File: docs/server-production-deploy.md [hunk 7]
```diff
@@ -49,17 +49,61 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+If you have the time, we recommend [watching our 19 minute video guide on YouTube](https://youtu.be/PqcVKIxI0nU) which demonstrates the debugging explained below.
+
+#### Basic Debugging on Temporal Web
+
+The primary mechanism we recommend for debugging is [Temporal Web](https://github.com/temporalio/web), which is run in a separate process:
+
+![6XkjmR](https://user-images.githubusercontent.com/6764957/110544958-71746480-8167-11eb-8152-8d3a3eb73d4e.gif)
+
+- [Workflows](https://docs.temporal.io/docs/glossary/#workflow) are identified by their [**Workflow ID**](https://docs.temporal.io/docs/glossary/#workflow-id), which you provide when creating the workflow. They also have a **Name** which is directly taken from your code.
+- Workflow ID's are are distinct from **Run ID's**, which uniquely identify one of potentially many runs of the same Workflow.
```
Comment (sw-yx) [inline docs/server-production-deploy.md:65]:
```suggestion
- Workflow ID's are are distinct from **Run ID's**, which uniquely identify one of potentially many runs of the same Workflow (don't confuse Runs with [Workflow Executions](https://docs.temporal.io/docs/glossary/#workflow-execution) - they are similar, but a single Workflow Execution can have multiple Runs. A Run is the atomic unit.)
```

File: docs/server-production-deploy.md [hunk 8]
```diff
@@ -49,17 +49,61 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+#### Basic Debugging on Temporal Web
+
+The primary mechanism we recommend for debugging is [Temporal Web](https://github.com/temporalio/web), which is run in a separate process:
+
+![6XkjmR](https://user-images.githubusercontent.com/6764957/110544958-71746480-8167-11eb-8152-8d3a3eb73d4e.gif)
+
+- [Workflows](https://docs.temporal.io/docs/glossary/#workflow) are identified by their [**Workflow ID**](https://docs.temporal.io/docs/glossary/#workflow-id), which you provide when creating the workflow. They also have a **Name** which is directly taken from your code.
+- Workflow ID's are are distinct from **Run ID's**, which uniquely identify one of potentially many runs of the same Workflow (don't confuse Runs with [Workflow Executions](https://docs.temporal.io/docs/glossary/#workflow-execution) - they are similar, but a long-running Workflow Execution can have multiple Runs. A Run is the atomic unit.)
+- Workflow **Status** is usually in one of a few states: Running, Completed, or Terminated, with **Start Time** and **End Time** shown accordingly.
+
```
Comment (sw-yx) [inline docs/server-production-deploy.md:67]:
```suggestion
- Workflow **Status** is usually in one of a few states: Running, Completed, or Terminated, with **Start Time** and **End Time** shown accordingly.
- Workflow ID's are are distinct from **Run ID's**, which uniquely identify one of potentially many Runs of the same Workflow.

> Tip: Don't confuse Runs with [Workflow Executions](https://docs.temporal.io/docs/glossary/#workflow-execution) - they are similar, but a long-running Workflow Execution can have multiple Runs. A Run is the atomic unit.
```

File: docs/server-production-deploy.md [hunk 9]
```diff
@@ -49,17 +49,62 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+#### Execution Histories on Temporal Web
+
+Reading execution histories is one of the more reliable ways of debugging:
+
+![image](https://user-images.githubusercontent.com/6764957/110546362-54d92c00-8169-11eb-81a6-74817e0d1378.png)
+
+Here, you can see the exact sequence of events that have happened so far, together with the relevant state for each event, and reason about what went wrong or what is preventing the next correct event. *There are about 40 system events in total - We intend to publish a full guide to each.*
+#### Viewing Stack Traces on Temporal Web
+
```
Comment (sw-yx) [inline docs/server-production-deploy.md:84]:
```suggestion
Here, you can see the exact sequence of events that have happened so far, together with the relevant state for each event, and reason about what went wrong or what is preventing the next correct event. *There are about 40 system events in total - We intend to publish a full guide to each.*

#### Viewing Stack Traces on Temporal Web

```

File: docs/server-production-deploy.md [hunk 10]
```diff
@@ -49,17 +49,63 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+#### Basic Debugging on Temporal Web
+
+The primary mechanism we recommend for debugging is [Temporal Web](https://github.com/temporalio/web), which is run in a separate process:
+
+![6XkjmR](https://user-images.githubusercontent.com/6764957/110544958-71746480-8167-11eb-8152-8d3a3eb73d4e.gif)
+
+- [Workflows](https://docs.temporal.io/docs/glossary/#workflow) are identified by their [**Workflow ID**](https://docs.temporal.io/docs/glossary/#workflow-id), which you provide when creating the workflow. They also have a **Name** which is directly taken from your code.
+- Workflow **Status** is usually in one of a few states: Running, Completed, or Terminated, with **Start Time** and **End Time** shown accordingly.
+- Workflow ID's are are distinct from **Run ID's**, which uniquely identify one of potentially many Runs of the same Workflow.
```
Comment (rylandg) [inline docs/server-production-deploy.md:66]:
```suggestion
- Workflow ID's are are distinct from **Run ID's**, which uniquely identify one of potentially many Runs of Workflows with the same Workflow ID.
```

File: docs/server-production-deploy.md [hunk 11]
```diff
@@ -49,17 +49,63 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+
+This is often a good way to get a deep understanding of whether your workflow is executing as expected.
+
+#### Recovering In-flight Workflows While Running
+
+Here we will discuss how to proceed oince you have identified and fixed the code for an erroring activity.
+
+If your activity code is deterministic, you might be able to simply restart the worker to pick up the changes. Execution will continue from where it last succeeded (typically an `ActivityTaskScheduled` event, without a corresponding `ActivityTaskCompleted`). In other words, we get "hotfixing for free" due to Temporal's execution model.
+
+However, if your activity is more complex/non-deterministic, you will have to use [our versioning feature](https://docs.temporal.io/docs/go-versioning/) or even manually terminate and restart the workflows.
```
Comment (rylandg) [inline docs/server-production-deploy.md:98]: Activity can be non-deterministic even for hot fixing. The complexity comes from when workflow code needs to be changed.

File: docs/server-production-deploy.md [hunk 12]
```diff
@@ -49,17 +49,63 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+
+![image](https://user-images.githubusercontent.com/6764957/110547621-20ff0600-816b-11eb-84f3-c6a97c5cad31.png)
+
+This is often a good way to get a deep understanding of whether your workflow is executing as expected.
+
+#### Recovering In-flight Workflows While Running
+
+Here we will discuss how to proceed oince you have identified and fixed the code for an erroring activity.
+
+If your activity code is deterministic, you might be able to simply restart the worker to pick up the changes. Execution will continue from where it last succeeded (typically an `ActivityTaskScheduled` event, without a corresponding `ActivityTaskCompleted`). In other words, we get "hotfixing for free" due to Temporal's execution model.
```
Comment (rylandg) [inline docs/server-production-deploy.md:96]:
```suggestion
If your activity code is deterministic, you might be able to simply restart the worker to pick up the changes. Execution will continue from where it last succeeded. In other words, we get "hotfixing for free" due to Temporal's execution model.
```

I would remove that

**Review by rylandg — APPROVED** [review]
Overall this is awesome. Thank you so much for doing this.

File: docs/server-production-deploy.md [hunk 13]
```diff
@@ -49,17 +49,63 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+
+However, if your activity is more complex/non-deterministic, you will have to use [our versioning feature](https://docs.temporal.io/docs/go-versioning/) or even manually terminate and restart the workflows.
```
Comment (sw-yx) [inline docs/server-production-deploy.md:98]:
```suggestion
However, if your activity is more complex, you will have to explicitly [version your workflows](https://docs.temporal.io/docs/go-versioning/) or even manually terminate and restart the workflows.
```

**Comment by sw-yx** [comment]
wanna make sure this passes your sniff test @flossypurse

File: docs/server-production-deploy.md [hunk 14]
```diff
@@ -49,17 +49,63 @@ Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_termina
...
+
+![6XkjmR](https://user-images.githubusercontent.com/6764957/110544958-71746480-8167-11eb-8152-8d3a3eb73d4e.gif)
+
+- [Workflows](https://docs.temporal.io/docs/glossary/#workflow) are identified by their [**Workflow ID**](https://docs.temporal.io/docs/glossary/#workflow-id), which you provide when creating the workflow. They also have a **Name** which is directly taken from your code.
+- Workflow **Status** is usually in one of a few states: Running, Completed, or Terminated, with **Start Time** and **End Time** shown accordingly.
+- Workflow ID's are are distinct from **Run ID's**, which uniquely identify one of potentially many Runs of Workflows with the same Workflow ID.
+
+> Tip: Don't confuse Runs with [Workflow Executions](https://docs.temporal.io/docs/glossary/#workflow-execution) - they are similar, but a long-running Workflow Execution can have multiple Runs. A Run is the atomic unit.
+The full state of every Run is inspectable in Temporal Web:
+
```
Comment (sw-yx) [inline docs/server-production-deploy.md:70]:
```suggestion
> Tip: Don't confuse Runs with [Workflow Executions](https://docs.temporal.io/docs/glossary/#workflow-execution) - they are similar, but a long-running Workflow Execution can have multiple Runs. A Run is the atomic unit.


The full state of every Run is inspectable in Temporal Web:
```

**Review by flossypurse — APPROVED** [review]
If you remember to do so, move each sentence to a new line.
Otherwise, lgtm for now.

**Comment by sw-yx** [comment]
gotcha - as you know i agree with that generally but dont really apply it to EVERY new sentence.

**Pull request merged**